probe_src/performance_test.py

```diff
@@ -1,190 +1,224 @@
 import subprocess
 import datetime
 import csv
-import psutil
 import time
+import os
+import shutil
+import resource
 from dataclasses import dataclass
+import psutil 
 
 @dataclass
 class Result:
     returncode: int
     cpu_times: tuple
-    memory_info: tuple
-    io_counters: tuple
+    memory_usage: int
+    io_counters: dict
     stdout: str
     stderr: str
     start_time: datetime.datetime
     end_time: datetime.datetime
-
-def benchmark_command(command: str, warmup_iterations: int, benchmark_iterations: int) -> list[Result]:
+    cpu_usage_percent: float
+
+LOG_FILE = "probe_log"
+RECORD_DIR = "probe_record"
+
+def cleanup():
+    if os.path.exists(LOG_FILE):
+        print("Removing log file.")
+        subprocess.run("rm probe_log", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if os.path.exists(RECORD_DIR):
+        print("Removing record directory.")
+        shutil.rmtree(RECORD_DIR)
+    time.sleep(3)
+
+def get_process_stats(proc):
+    try:
```

Comment:
Let's try using [this](https://stackoverflow.com/questions/26475636/measure-elapsed-time-amount-of-memory-and-cpu-used-by-the-extern-program) method to capture the CPU times (just make sure to include a link to the StackOverflow in a comment). I didn't notice that psutil doesn't provide this functionality natively.

I hope that solves the "cpu time is always zero" issue.

Reply:
I have currently used `getrusage` in this commit since we just had one child process, it returned non-zero values. The stackoverflow method also mentions that if there is just one child process we can use `getrusage` . should i try the other method mentioned in the stackoverflow article (`subprocess.Popen`) ?

Reply:
There will be more than one child process in some important test-cases (e.g., cmd-in-bash), so we should switch to the solution which subclasses `subprocess.Popen`. In fact, `resource_call` in that answer should also measure wall time and return a `Result`.